src/function.rs

```diff
@@ -114,6 +121,10 @@ fn clean(_context: &FunctionContext, path: &str) -> Result<String, String> {
   Ok(Path::new(path).lexiclean().to_str().unwrap().to_owned())
 }
 
+fn dirs_map(f: fn() -> Option<PathBuf>) -> Result<String, String> {
+  Ok(f().map_or(String::new(), |p| p.to_string_lossy().into_owned()))
```

Comment:
We should use String::from_utf8, and return an error if conversion fails. `to_string_lossy` will replace invalid UTF-8 sequences with replacement characters, if the user subsequently tries to access the returned path, it will fail.

We should also add a `&'static str` argument to `dirs_map`, which is the name of the directory "cache directory", etc, we're looking up, and include it in the error message.

Reply:
Also, let's call `dirs_map` `dir`.

Reply:
Thanks for the review! I made this change